app/lib/pdf.js

```diff
@@ -35,35 +36,52 @@ const DEFAULTS = {
  * @param {PdfGetOptions} [options] - PDF options
  * @return { Promise } a promise that returns the PDF
  */
-async function get(url, options = {}) {
+async function get(
+    url,
+    {
+        format = DEFAULTS.FORMAT,
+        margin = DEFAULTS.MARGIN,
+        landscape = DEFAULTS.LANDSCAPE,
+        scale = DEFAULTS.SCALE,
+    } = {}
+) {
     if (!url) {
         throw new Error('No url provided');
     }
 
-    options.format = options.format || DEFAULTS.FORMAT;
-    options.margin = options.margin || DEFAULTS.MARGIN;
-    options.landscape = options.landscape || DEFAULTS.LANDSCAPE;
-    options.scale = options.scale || DEFAULTS.SCALE;
-
     const urlObj = new URL(url);
-    urlObj.searchParams.append('format', options.format);
-    urlObj.searchParams.append('margin', options.margin);
-    urlObj.searchParams.append('landscape', options.landscape);
-    urlObj.searchParams.append('scale', options.scale);
+    urlObj.searchParams.append('format', format);
+    urlObj.searchParams.append('margin', margin);
+    urlObj.searchParams.append('landscape', landscape);
+    urlObj.searchParams.append('scale', scale);
 
-    const browser = await puppeteer.launch({ headless: true });
+    const browser = await getBrowser(browserHandler);
     const page = await browser.newPage();
 
     let pdf;
 
     try {
-        await page
+        // To use an eventhandler here and catch a specific error,
+        // we have to return a Promise (in this case one that never resolves).
+        const detect401 = new Promise((resolve, reject) => {
+            page.on('requestfinished', (request) => {
+                if (request.response().status() === 401) {
+                    const e = new Error('Authentication required');
+                    e.status = 401;
+                    reject(e);
+                }
+            });
+        });
+        const goToPage = page
             .goto(urlObj.href, { waitUntil: 'networkidle0', timeout })
             .catch((e) => {
                 e.status = /timeout/i.test(e.message) ? 408 : 400;
                 throw e;
             });
 
+        // Either a 401 error is thrown or goto succeeds (or encounters a real loading error)
+        await Promise.race([detect401, goToPage]);
+
```

Comment:
As you can see, it was a challenge to properly catch a 401 response... I would certainly be impressed if y'all can point to a better solution :).